validator/gke_restrict_pod_traffic_test.rego

```diff
@@ -0,0 +1,62 @@
+#
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package templates.gcp.GKERestrictPodTrafficConstraintV1
+
+import data.validator.gcp.lib as lib
+
+all_violations[violation] {
+	resource := data.test.fixtures.assets.gke_restrict_pod_traffic[_]
+	constraint := data.test.fixtures.constraints.restrict_gke_pod_traffic
+
+	issues := deny with input.asset as resource
+		 with input.constraint as constraint
+
+	violation := issues[_]
+}
+
+good_config_no_violations[violation] {
+	resource := data.test.fixtures.assets.gke_restrict_pod_traffic[_]
+	constraint := data.test.fixtures.constraints.restrict_gke_pod_traffic
+
+	resource.name = "//container.googleapis.com/projects/transfer-repos/zones/us-central1-c/clusters/joe-clust4"
+
+	issues := deny with input.asset as resource
+		 with input.constraint as constraint
+
+	violation := issues[_]
+}
+
+test_violations_basic {
+	# 3 violation cases in the test fixtures:
+	#   1. network_policy_config disabled
+	#   2. network_policy not exists
```

Comment:
nit: "does not exist"

Reply:
can we add a test case where it exists and is explicitly set to false?

Reply:
Updated